factory_decorator_type_check.py

```diff
@@ -0,0 +1,108 @@
+from functools import wraps
+
+
+class CustomError(Exception):
```

Comment:
I asked you to create a custom exception - it means create your own exception\raise a good exception. 
Always throw the most specific exception - 
https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python
https://docs.python.org/3/library/exceptions.html